app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.android.ui.home.gallery
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.model.ImageAsset
+import com.wire.android.navigation.EXTRA_IMAGE_DATA
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.navigation.parseIntoPrivateImageAsset
+import com.wire.android.util.ui.WireSessionImageLoader
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MediaGalleryViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    val navigationManager: NavigationManager,
+    val wireSessionImageLoader: WireSessionImageLoader,
+    val getConversationDetails: ObserveConversationDetailsUseCase
+) : ViewModel() {
+
+    var mediaGalleryViewState by mutableStateOf(MediaGalleryViewState())
+        private set
+
+    val imageAssetId: ImageAsset.PrivateAsset = savedStateHandle.get<String>(EXTRA_IMAGE_DATA)!!.parseIntoPrivateImageAsset()
```

Comment:
I wonder if this null force unwrap can trigger a NPE 🤔  What do you think of using a `?` instead of `!!` ?
Just asking out of curiosity, but if We know for sure that this case can't happen, just ignore my comment :D

Reply:
This is similar to what we do on `ConversationViewModel`,  where we parse the `conversationId` received via the arguments route. If this parameter is null, then imo it is fair to throw the NPE so that we detect that something else went wrong.

Reply:
If we decide not to, how should we handle the situation there? just navigate to previous screen?